docs: add ROADMAP.md (#2579)
microsoft/rnx-kit: ROADMAP.md

  development client that can be downloaded; for instance, from the platform's
  app store or as a downloadable executable.
- Where access to the native code or additional dependencies are required, we
  need to make it effortless to include additional modules and to change React
  Native versions (in both directions).

This effort predates `rnx-kit`, and we've achieved many of the things we've
aimed for with [`react-native-test-app`][], but still need the last push over
the finishing line.

For historical reasons, this work is in a separate repository:
https://github.com/microsoft/react-native-test-app. It is maintained by the very
same folks maintaining `rnx-kit`. It currently does not support/target web.

In the long term, we want `react-native-test-app` to be the best choice for
developers, whether they are working on a library or a standalone app.

> **Note**
>
> - For efforts towards making it easier to build, see
>   [Building native](#building-native).
>
> Other work we are aware of in this area:
>
> - Shopify created [Tophat][] to improve their inner dev loop by leveraging CI
>   and local tooling to allow seamless local testing of PR builds (iOS and
>   Android only).

> ### React Native Web APIs

Reviewer: ```suggestion
### React Native WebAPIs
```

The piece of the browser puzzle missing from the above section is a standard
API. Today, web developers can reach out to Web APIs and assume that their app
will function across many platforms — that is not the case for React Native
developers. If the API you're looking for is not immediately available in
`react-native`, you will have to look for modules written by the community.
Maybe you'll find one in [React Native Directory][], or maybe you will have to
write one yourself. We want to shift the mindset from "what packages to use" to
"what APIs are available".

Standard APIs is what will make the "web browser" effort truly ubiquitous. We
cannot provide a complete downloadable app for development without this.

> **Note**
>

> > - The RFC for this effort is here: [React Native WebAPI implementation][]

Reviewer: ```suggestion
> - The RFC for this effort is here: [React Native WebAPIs][]
```

                      </li>
                      <li>Java Development Kit</li>
                    </ul>
                  </li>
                  <li>
                    iOS
                    <ul>
                      <li>Xcode</li>
                    </ul>
                  </li>
                  <li>
                    macOS
                    <ul>
                      <li>Xcode (only needed for native debugging)</li>
                    </ul>
                  </li>
                  <li>
                    Windows
                    <ul>
                      <li>Visual Studio (only needed for native debugging)</li>
                    </ul>
                  </li>
                </ul>
              </li>
            </ul>
          </li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>API</td>
      <td>
        <a href="https://developer.mozilla.org/en-US/docs/Web/API">Web APIs</a>
      </td>
      <td>
        The React Native implementation of the very same
        <a href="https://developer.mozilla.org/en-US/docs/Web/API">Web APIs</a>
        as detailed in the
        <a href="https://github.com/microsoft/rnx-kit/pull/2504"

>           >React Native WebAPI implementation RFC</a

Reviewer: ```suggestion
          >React Native WebAPIs RFC</a
```

<!-- References -->

[`@react-native-community/cli`]: https://github.com/react-native-community/cli
[`@rnx-kit/align-deps`]:
  https://github.com/microsoft/rnx-kit/tree/main/packages/align-deps#readme
[`@rnx-kit/build`]:
  https://github.com/microsoft/rnx-kit/tree/main/incubator/build#readme
[`npm create vite@latest`]:
  https://vitejs.dev/guide/#scaffolding-your-first-vite-project
[`react-native-test-app`]: https://github.com/microsoft/react-native-test-app
[Bundle Working Group]:
  https://github.com/microsoft/rnx-kit/discussions/categories/bundle-working-group
[Chrome DevTools]: https://developer.chrome.com/docs/devtools/
[Debugging Working Group]:
  https://github.com/react-native-community/developer-experience-wg/discussions/categories/debugging
[GitHub Codespaces]: https://github.com/features/codespaces
[Metro]: https://facebook.github.io/metro/
[Metro Office Hours]:
  https://github.com/react-native-community/developer-experience-wg/discussions/categories/metro
[React DOM For Native RFC]:
  https://github.com/react-native-community/discussions-and-proposals/pull/496
[React Native Directory]: https://reactnative.directory/

> [React Native WebAPI implementation]:

Reviewer: ```suggestion
[React Native WebAPIs]:
```
